### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry): quasi-compact + specializing => closed map

### PR DESCRIPTION
**Pull request by erdOne**
Partially from the valuative criterion project

Co-authored-by: Christian Merten

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/noshake.json [hunk 1]
```diff
@@ -408,6 +408,8 @@
   ["Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Unital"],
   "Mathlib.AlgebraicTopology.DoldKan.Notations":
   ["Mathlib.AlgebraicTopology.AlternatingFaceMapComplex"],
+  "Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic":
+  ["Mathlib.Tactic.StacksAttribute"],
```
Comment (YaelDillies) [inline scripts/noshake.json:412]: Shouldn't `Tactic.StacksAttribute` be made one of those global noshakes? See `ignoreImport` at the top of this file

**Review by YaelDillies — APPROVED** [review]
Looks good to me style-wise, and also maths-wise but I am not an algebraic geometer

File: scripts/noshake.json (hunk 1 above)
Comment (jcommelin) [reply scripts/noshake.json:412]: Sounds good to me.

**Review by jcommelin — COMMENTED** [review]
LGTM, but I like Yael's suggestion about the noshake.

Please fix that and merge afterwards.

bors d+

**Comment by erdOne** [comment]
bors merge

File: Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean [hunk 2]
```diff
@@ -534,6 +535,86 @@ def localizationMapOfSpecializes {x y : PrimeSpectrum R} (h : x ⤳ y) :
...
+  have : (comap f '' zeroLocus I) = Set.range (comap ((Ideal.Quotient.mk I).comp f)) := by
+    rw [comap_comp, ContinuousMap.coe_comp, Set.range_comp, range_comap_of_surjective, Ideal.mk_ker]
+    exact Ideal.Quotient.mk_surjective
+  rw [this] at hf ⊢
+  exact isClosed_range_of_stableUnderSpecialization _ hf
+
+variable {f} in
+@[stacks 05JL]
+lemma stableUnderSpecialization_range_iff :
+    StableUnderSpecialization (Set.range (comap f)) ↔ IsClosed (Set.range (comap f)) :=
```
Comment (alreadydone) [inline Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean:568]: These are closer to [00HY](https://stacks.math.columbia.edu/tag/00HY) which is about prime spectra, while [05JL](https://stacks.math.columbia.edu/tag/05JL) is about schemes. I'm changing the labels in #20533.

**Pull request closed without merging**